app/routes/planet_routes.py

```diff
@@ -0,0 +1,93 @@
+from flask import Blueprint, abort, make_response, request, Response
+from ..models.planet import Planet
+from ..db import db
+from sqlalchemy import asc, desc
+
+planets_bp = Blueprint("planets_bp", __name__, url_prefix="/planets")
+
+@planets_bp.post("")
+def create_planet():
+    request_body = request.get_json()
+    name = request_body["name"]
+    surface_area = request_body["surface_area"]
+    moons = request_body["moons"]
+    distance_from_sun = request_body["distance_from_sun"]
+    namesake = request_body["namesake"]
+
+    new_planet = Planet(name=name, surface_area=surface_area, moons=moons, distance_from_sun=distance_from_sun, namesake=namesake)
+    db.session.add(new_planet)
+    db.session.commit()
+
+    response = new_planet.to_dict()
+    return response, 201
+
+@planets_bp.get("")
+def get_all_planets():
+    sort_param = request.args.get("sort")
+    order_param = request.args.get("order", "asc") # Default to ascending if not specified
+
+    query = db.select(Planet)
+
+    if sort_param:
+        sort_column = getattr(Planet, sort_param, None)
+        if sort_column:
+            query = query.order_by(asc(sort_column) if order_param == "asc" else desc(sort_column))
+
+    namesake_param = request.args.get("namesake")
+    if namesake_param:
+        query = query.where(Planet.namesake.ilike(f"%{namesake_param}%"))
+
+    distance_from_sun_param = request.args.get("distance_from_sun")
+    if distance_from_sun_param:
+        query = query.where(Planet.distance_from_sun <= distance_from_sun_param)
+    query = query.order_by(Planet.id)
+
+    planets = db.session.scalars(query)
```

Comment:
Since adding the `order_by` isn't tightly related to filtering by `distance_from_sun_param`, I'd consider adding space between those statements:
```py
    distance_from_sun_param = request.args.get("distance_from_sun")
    if distance_from_sun_param:
        query = query.where(Planet.distance_from_sun <= distance_from_sun_param)
    
    query = query.order_by(Planet.id)
    planets = db.session.scalars(query)
```